### PR TITLE
added chrome.declarativeContent.SetIcon class

### PR DESCRIPTION
**Pull request by bokenator**
https://developer.chrome.com/extensions/declarativeContent#type-SetIcon

### CODE REVIEW

File: types/chrome/index.d.ts [hunk 1]
```diff
@@ -1593,7 +1593,14 @@ declare namespace chrome.declarativeContent {
     }
 
     /** Declarative event action that shows the extension's page action while the corresponding conditions are met. */
-    export class ShowPageAction { }
+	export class ShowPageAction { }
```
Comment (danger-public) [inline types/chrome/index.d.ts:1596]: - :no_entry_sign: The root .editorconfig style specifies spaces for whitespace. Please use spaces in new or changed types.

File: types/chrome/index.d.ts [hunk 2]
```diff
@@ -1595,6 +1595,13 @@ declare namespace chrome.declarativeContent {
     /** Declarative event action that shows the extension's page action while the corresponding conditions are met. */
     export class ShowPageAction { }
 
+    /** Declarative event action that changes the icon of the page action while the corresponding conditions are met. */
+    export class SetIcon {
+        constructor (options: {
```
Comment (peterblazejewicz) [inline types/chrome/index.d.ts:1600]:
```suggestion
        constructor (options?: {
```
Docs seems to says that's optional (`(optional) imageData`). Is that correct?

File: types/chrome/test/index.ts [hunk 3]
```diff
@@ -398,6 +398,26 @@ function testDebugger() {
 	});
 }
 
+// https://developer.chrome.com/extensions/declarativeContent
+function testDeclarativeContent() {
+    let activeIcon: ImageData = new ImageData(32, 32);
```
Comment (peterblazejewicz) [inline types/chrome/test/index.ts:403]:
```suggestion
    const activeIcon: ImageData = new ImageData(32, 32);
```
This is old DT package, not following strictly current checks, `const` is preffered now

File: types/chrome/test/index.ts [hunk 4]
```diff
@@ -398,6 +398,26 @@ function testDebugger() {
...
+
+    let rule: chrome.events.Rule = {
```
Comment (peterblazejewicz) [inline types/chrome/test/index.ts:405]:
```suggestion
    const rule: chrome.events.Rule = {
```

**Review by peterblazejewicz — APPROVED** [review]
LGTM!
@bokenator thanks for the contribution!

**Review by peterblazejewicz — APPROVED** [review]
:shipit:

File: types/chrome/index.d.ts [hunk 5]
```diff
@@ -1595,6 +1595,13 @@ declare namespace chrome.declarativeContent {
     /** Declarative event action that shows the extension's page action while the corresponding conditions are met. */
     export class ShowPageAction { }
 
+    /** Declarative event action that changes the icon of the page action while the corresponding conditions are met. */
+    export class SetIcon {
+        constructor (options?: {
```
Comment (sheetalkamat) [inline types/chrome/index.d.ts:1600]: From the document it seems 
`Either an ImageData object or a dictionary {size -> ImageData} representing an icon to be set` so should this be `constructor (options?: ImageData | { [size: string]: ImageData })` instead

**Review by sheetalkamat — CHANGES_REQUESTED** [review]

File: types/chrome/index.d.ts (hunk 5 above)
Comment (bokenator) [reply types/chrome/index.d.ts:1600]: The dictionary `{size -> ImageData}` is still passed in as a part of the options object:
```
new SetIcon({
imageData: {
'16': iconImageData
}
});
```

**Pull request merged**